Source/View Controllers/UploadViewController.swift

```diff
@@ -230,22 +245,26 @@ class UploadViewController: NSViewController {
     }
 
     func updateCredentialsAvailable() {
-        guard credentialsAvailable != (
-            !jamfProServerLabel.stringValue.isEmpty
-            && (username != nil)
-            && !username.isEmpty
-            && (password != nil)
-            && !password.isEmpty) else { return }
-        credentialsAvailable = !credentialsAvailable
+        guard let username = username, !username.isEmpty,
+              let password = password, !password.isEmpty,
+              !jamfProServerLabel.stringValue.isEmpty else {
+            if credentialsAvailable { credentialsAvailable = false }
+            return
+        }
+
+        if !credentialsAvailable { credentialsAvailable = true }
     }
 
     func updateReadForUpload() {
+        guard let payloadName = payloadName, !payloadName.isEmpty else {
+            if readyForUpload { readyForUpload = false}
```

Comment:
Just settings `readyForUpload` is good enough, I think.

```suggestion
            readyForUpload = false
```

Reply:
Changed with latest commit